jdatetime/__init__.py

```diff
@@ -43,6 +43,30 @@
     FA_LOCALE = 'fa_IR'
 
 
+def _format_time(hour, minute, second, microsecond, timespec='auto'):
+    specs = {
+        'hours': '{:02d}',
+        'minutes': '{:02d}:{:02d}',
+        'seconds': '{:02d}:{:02d}:{:02d}',
+        'milliseconds': '{:02d}:{:02d}:{:02d}.{:03d}',
+        'microseconds': '{:02d}:{:02d}:{:02d}.{:06d}',
+    }
+
+    if timespec == 'auto':
+        # Skip trailing microseconds when equals to 0
+        timespec = 'microseconds' if microsecond else 'seconds'
+    elif timespec == 'milliseconds':
+        # convert to millisecond
+        microsecond //= 1000
+
+    try:
+        fmt = specs[timespec]
+    except KeyError:
+        raise ValueError('Unknown timespec value: %s' % timespec)
+    else:
```

Comment:
Is this `else` here really needed? Because of the `raise` statement in the `except` block, the only code path after the `try`/`except` is the valid case where we can `return`.

Reply:
that is in fact how the [original library](https://github.com/python/cpython/blob/e226e83d36dfc7220d836fb7a249ce18e70cb4a6/Lib/datetime.py#L174:L175) works:
```
    else:
        return fmt.format(hh, mm, ss, us)
```

but are you suggesting that they are doing it all wrong too?

Reply:
> Is this else here really needed?

not wrong, nor necessary. The question was to learn the reason behind this choice, and the answer is now clear.